test/pathofthetemplarshard.ts

```diff
@@ -0,0 +1,125 @@
+import { ethers } from "hardhat";
+import { TypedDataDomain, TypedDataField, TypedDataSigner } from "@ethersproject/abstract-signer";
+import { expect } from "chai";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import {
+  PathOfTheTemplarShard, PathOfTheTemplarShard__factory,
+  Relic, Relic__factory,
+  Shards, Shards__factory,
+} from "../typechain";
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import path from "path";
+
+type MySigner = SignerWithAddress & TypedDataSigner;
+
+let owner: MySigner;
+let add2: MySigner;
+let add3: MySigner;
+let add4: MySigner;
+let pathOfTheTemplarShard: PathOfTheTemplarShard;
+let relic: Relic;
+let shards: Shards;
+let SHARD_ID: number[] = [2, 3, 4, 5, 6];
+let ENCLAVE: string[] = [
+    "",
+    "chaosEnclave",
+    "mysteryEnclave",
+    "logicEnclave",
+    "structureEnclave",
+    "orderEnclave"
+];
+
+describe("PathOfTheTemplarShard", async () => {
+
+    before( async () => {
+        [owner, add2, add3, add4] = await ethers.getSigners();
+    });
+
+    async function setup() {
+        relic = await new Relic__factory(owner).deploy();
+        shards = await new Shards__factory(owner).deploy();
+        pathOfTheTemplarShard = await new PathOfTheTemplarShard__factory(owner).deploy();
+
+        return {
+            relic,
+            shards,
+            pathOfTheTemplarShard,
+        }
+    }
+
+    beforeEach( async () => {
+        ({
+            relic,
+            shards,
+            pathOfTheTemplarShard,
+        } = await loadFixture(setup));
+    });
+
+    it("Check if deployer is owner", async function () {
+        expect(await pathOfTheTemplarShard.owner()).to.equal(await owner.getAddress());
+    });
+
+    it("Check if mapping of enclave to Shard Id is correct", async function () {
+        await pathOfTheTemplarShard.establishMapping();
+    
+        for (let i = 2; i < SHARD_ID.length; i++) {
+          expect(await pathOfTheTemplarShard.getEnclaveForShard(SHARD_ID[i])).to.equal(ENCLAVE[i]);
+        }
+    });
+
+    it("Check if owner can set Minter Role", async function () {
+        const account = await add2.getAddress();
+        const value = true;
+      
+        await expect(
+            pathOfTheTemplarShard.connect(owner).setMinter(account, value)
+        )
+            .to.emit(pathOfTheTemplarShard, 'MinterSet')
+            .withArgs(account, value);
+      
+        expect(await pathOfTheTemplarShard.minters(account)).to.equal(value);
+
+      });
+
+      it("Check if a signer that is not whitelisted minter produces an invalid signature", async function () {
```

Comment:
get signed typed data of address which can't mint and expect custom error from contract after calling with:
- invalid signature
- can't mint address